File: CloudflareFileStorage.cs
using System;
using System.IO;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Threading.Tasks;

public class CloudflareFileStorage
{
    private readonly HttpClient _httpClient;
    private readonly string _workerUrl; // Cloudflare Worker URL

    public CloudflareFileStorage(string workerUrl)
    {
        _httpClient = new HttpClient();
        _workerUrl = workerUrl.TrimEnd('/');
    }

    public async Task<string> UploadFileAsync(string filePath)
    {
        if (!File.Exists(filePath))
            throw new FileNotFoundException("File not found.", filePath);

        using var content = new MultipartFormDataContent();
        using var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read);
        var fileContent = new StreamContent(fileStream);
        fileContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/octet-stream");
        content.Add(fileContent, "file", Path.GetFileName(filePath));

        var response = await _httpClient.PostAsync($"{_workerUrl}/upload", content);
        return await response.Content.ReadAsStringAsync();
    }

    public async Task<byte[]> DownloadFileAsync(string fileName, string savePath)
    {
        var response = await _httpClient.GetAsync($"{_workerUrl}/download/{fileName}");
        if (!response.IsSuccessStatusCode)
            throw new Exception($"Error downloading file: {response.ReasonPhrase}");

        var fileBytes = await response.Content.ReadAsByteArrayAsync();
        await File.WriteAllBytesAsync(Path.Combine(savePath, fileName), fileBytes);
        return fileBytes;
    }

    public async Task<string> DeleteFileAsync(string fileName)
    {
        var response = await _httpClient.DeleteAsync($"{_workerUrl}/delete/{fileName}");
        return await response.Content.ReadAsStringAsync();
    }
}

File: CloudflareFileStorage.php
<?php

class CloudflareFileStorage
{
    private $workerUrl;

    public function __construct($workerUrl)
    {
        $this->workerUrl = rtrim($workerUrl, '/');
    }

    public function uploadFile($filePath)
    {
        if (!file_exists($filePath)) {
            throw new Exception("File not found: $filePath");
        }

        $file = new CURLFile($filePath, 'application/octet-stream', basename($filePath));
        $data = ['file' => $file];
        $ch = curl_init("{$this->workerUrl}/upload");
        curl_setopt($ch, CURLOPT_POST, true);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $data);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        $response = curl_exec($ch);
        curl_close($ch);

        return $response;
    }

    public function downloadFile($fileName, $savePath)
    {
        $ch = curl_init("{$this->workerUrl}/download/$fileName");
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        $fileContent = curl_exec($ch);
        curl_close($ch);

        if ($fileContent === false) {
            throw new Exception("Error downloading file.");
        }

        file_put_contents($savePath . DIRECTORY_SEPARATOR . $fileName, $fileContent);
        return $fileContent;
    }

    public function deleteFile($fileName)
    {
        $ch = curl_init("{$this->workerUrl}/delete/$fileName");
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "DELETE");
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        $response = curl_exec($ch);
        curl_close($ch);

        return $response;
    }
}
?>
